lib/src/utils.dart

```diff
@@ -82,3 +63,74 @@ void logDependencyInfractions(String infraction, Iterable<String> dependencies)
   final sortedDependencies = dependencies.toList()..sort();
   logger.warning([infraction, bulletItems(sortedDependencies), ''].join('\n'));
 }
+
+/// Lists the packages with infractions
+List<String> getDependenciesWithPins(Map dependencies) {
+  final List<String> infractions = [];
+  for (String packageName in dependencies.keys) {
+    String version;
+    final packageMeta = dependencies[packageName];
+
+    if (dependencies[packageName] is String) {
+      version = packageMeta;
+    } else {
```

Comment:
Looks like this didn't get addressed:

> Should we have a check for `is Map` here? The Yaml spec is pretty complex and their may be some edge-cases here.

I think we should do this. I actually just ran into it when testing wdesk_sdk (lolwut):
https://github.com/Workiva/wdesk_sdk/blob/master/pubspec.yaml#L38

Reply:
Does empty string resolve as any?